airbyte-notification/src/main/java/io/airbyte/notification/SlackNotificationClient.java

```diff
@@ -0,0 +1,102 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.notification;
+
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.config.SlackNotificationConfiguration;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import org.apache.logging.log4j.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Notification client that uses Slack API for Incoming Webhook to send messages.
+ *
+ * This class also reads a resource YAML file that defines the template message to send.
+ *
+ * It is stored as a YAML so that we can easily change the structure of the JSON data expected by
+ * the API that we are posting to (and we can write multi-line strings more easily).
+ *
+ * For example, slack API expects some text message in the { "text" : "Hello World" } field...
+ */
+public class SlackNotificationClient implements NotificationClient {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SlackNotificationClient.class);
+
+  private final HttpClient httpClient = HttpClient.newBuilder()
+      .version(HttpClient.Version.HTTP_2)
+      .build();
+  private final SlackNotificationConfiguration config;
+
+  public SlackNotificationClient(final SlackNotificationConfiguration config) {
+    this.config = config;
+  }
+
+  @Override
+  public boolean notifyJobFailure(String sourceConnector, String destinationConnector, String jobDescription, String logUrl)
+      throws IOException, InterruptedException {
+    return notify(renderJobData(
+        "failure_slack_notification_template.txt",
+        sourceConnector,
+        destinationConnector,
+        jobDescription,
+        logUrl));
+  }
+
+  private String renderJobData(String templateFile, String sourceConnector, String destinationConnector, String jobDescription, String logUrl)
+      throws IOException {
+    final String template = MoreResources.readResource(templateFile);
```

Comment:
cool. not super obvious to me that you need to use a file at all as opposed to just having the string in here. up to you.